### DISCUSSION TITLE
Plugin ideas

### DISCUSSION THREAD

**Discussion by gmrchk** [swup/swup#335] [Feature ideas]
Do you have a plugin idea, or maybe a plugin ready that you'd like to share with others? Please, post below!

If you like anything, don't forget to leave a 👍 reaction, to less us know what's important for you guys.

**Comment by gmrchk** [comment]
I think the reCAPTCHA plugin is long overdue, and if I remember correctly, it has been requested a few times.

**Comment by daun** [comment]
A plugin for defining sub-sections or sub-frames would be great. A common use case is a `#content` container with a filter form, where you want the filter form to persist across page loads but want the `#result` container inside `#content` replaced instead of `#content` itself.

Something like this is currently possible by handling it yourself and adding custom history entries, but it'd be nice to have an officially sanctioned way of doing it.

**Reply by daun** [reply]
Update: we now have the [fragment plugin](https://github.com/swup/fragment-plugin) for replacing dynamic parts of the page 🎉

**Comment by scsskid** [comment]
a plugin thats applying from-[route] classes in addtition to to-[route] classes

**Reply by daun** [reply]
Update: we now have the [route name plugin](https://github.com/swup/route-name-plugin) that handles that and more.
